conftest.py

```diff
@@ -23,3 +29,23 @@ def pytest_addoption(parser):
 
 if sys.version_info < (3, 6):
     collect_ignore.append('pavement.py')
+
+
+@pytest.fixture(autouse=True)
+def global_tmpdir(request, tmpdir):
+    """
+    Return a temporary directory path object which is unique to each test
+    function invocation, created as a sub directory of the base temporary
+    directory. The returned object is a ``tests.lib.path.Path`` object.
+
+    This uses the built-in tmpdir fixture from pytest itself but modified
+    to return our typical path object instead of py.path.local as well as
+    deleting the temporary directories at the end of each test case.
+    """
+    assert tmpdir.isdir()
+    yield Path(str(tmpdir))
```

Comment:
You don't need to cast to `str` first, as `path.Path` is a subclass of `str`/`unicode`.